File: solutions/00001 Two Sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int,int> buff_dict;
        for (auto i = 0; i < nums.size(); ++i) {
            if (buff_dict.find(nums[i]) != buff_dict.end()){
                return {buff_dict[nums[i]],i};
            }
            buff_dict.emplace(target-nums[i], i);
        }
        return {};
    }
};

File: solutions/00001 Two Sum/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        buff_dict = {}
        for i, num in enumerate(nums):
            if num in buff_dict:
                return [buff_dict[num], i]
            buff_dict[target-num] = i
        return []
